test/integrationTests/unitTests.integration.test.ts

```diff
@@ -118,3 +150,15 @@ async function getCodeLensesAsync(): Promise<vscode.CodeLens[]> {
         return a.command!.title.localeCompare(b.command!.command);
     });
 }
+
+async function setConfigurationAndWaitForObserver<T>(configuration: string, value: T): Promise<void> {
+    const changed = new Promise<void>((resolve, _) => {
+        vscode.workspace.onDidChangeConfiguration((e) => {
```

Comment:
incredibly annoyingly, `optionProvider.GetLatestOptions()` does not actually read the latest options.  It instead uses whatever the observable last triggered with.  This is extremely racy.

It isn't simple to modify `GetLatestOptions()` because we don't want to be reading the entire option set every single time.  Instead I think I need to refactor options to instead return functions to get the latest value, instead of the entire set of options.

Reply:
filed https://github.com/dotnet/vscode-csharp/issues/6337 for this